### DISCUSSION TITLE
What is the difference between ref.watch() and ref.read()?

### DISCUSSION THREAD

**Discussion by kritinds** [community/community#28068] [Mobile] [answered]
What is the difference between ref.watch() and ref.read()?

**Comment by airtower-luna** [comment]
In most programming languages those look like two different method calls on an object. But to get a helpful answer please provide a bit more context: What programming language are you using, and (at least) as much code as necessary to see what your `ref` is. :smile_cat:

**Comment by sthakrishnakumar** [comment, marked as answer]
Hello @kritinds ,
                         these both read and watch have similar kind of functionality but the purpose of use is different.
we can use ref.read() to the get the value of provider just once i.e. one time reading ,
but in case of ref.watch() , it is used to get the value for the first time as of like ref.read() and also to get the value every single time it changes i.e. it provides every change of value. Also for example, ref.watch() is like you are subscribing a YouTube channel and you get notified when new content is added.
Hope it helps you .

**Reply by kritinds** [reply]
Thank you so much

**Comment by supervoid0** [comment]
**ref.read()**
Use to get the value in provider just once.

**ref.watch()**
Get the value in provider the first time and every time the value changes. Here ref.watch() observes changes in value.

So, if your goal is to get the value everytime the value changes, use ref.watch()

@kritinds

**Reply by kritinds** [reply]
thank you
